cjdns-hdr/src/switch_header.rs

```diff
@@ -0,0 +1,126 @@
+//! Logic for cjdns switch header parsing and serialization
+
+use cjdns_core::RoutingLabel;
+
+use crate::{
+    errors::{HeaderError, Result},
+    utils::{Reader, Writer},
+};
+
+const SWITCH_HEADER_SIZE: usize = 12;
+const HEADER_CURRENT_VERSION: u8 = 1;
```

Comment:
Can be moved into `impl` block with shorter names, like `SwitchHeader::SIZE` which can be reused in other submodule (see other comment).